utils/nativefuncs.go

```diff
@@ -242,6 +243,33 @@ func RegisterNativeFuncs(vm *jsonnet.VM, resolver Resolver) {
 			return ret, nil
 		},
 	})
+
+	vm.NativeFunction(&jsonnet.NativeFunction{
+		Name:   "validateJSONSchema",
+		Params: []jsonnetAst.Identifier{"obj", "schema"},
+		Func: func(args []interface{}) (interface{}, error) {
+			obj := args[0].(map[string]interface{})
+			schema := args[1].(map[string]interface{})
+
+			jSchema, err := json.Marshal(schema)
+			if err != nil {
+				return nil, fmt.Errorf("unable to json marshal schema: %w\n", err)
+			}
+
+			// No URL defaults to draft 7 of JSONSchema
+			sch, err := jsonschema.CompileString("", string(jSchema))
+			if err != nil {
+				return false, fmt.Errorf("unable to compile jsonschema: %w\n", err)
```

Comment:
Considering this means that the schema itself is invalid, i.e. a misplaced `required: [<fields>]` key I think this could also be `return nil, ...`.

Perhaps it should be, in order to avoid false negatives :thinking: 

**Thoughts?**

Reply:
yeah returning nil on error is ok.

The reason we return `true` on success (as opposed to null) is to make the call to validateJSONSchema "assert-friendly".

```jsonnet
{
  assert kubecfg.validateJSONSchema(self, import 'schema.json'),

  // ...
}

Reply:
Makes sense :+1: 

The `assert` explanation helps too, that formalises more what we discussed before. I'll swap this to `return nil, ...`